lib/alexa.js

```diff
@@ -1,418 +1,573 @@
-        value: context,
-        writable: false
-    });
-
-    Object.defineProperty(handler, '_callback', {
-        value: callback,
-        writable: false
-    });
-
-    Object.defineProperty(handler, 'state', {
-        value: null,
-        writable: true,
-        configurable: true
-    });
-
-    Object.defineProperty(handler, 'appId', {
-        value: null,
-        writable: true
-    });
-
-    Object.defineProperty(handler, 'response', {
-        value: {},
-        writable: true
-    });
-
-    Object.defineProperty(handler, 'dynamoDBTableName', {
-        value: null,
-        writable: true
-    });
-
-    Object.defineProperty(handler, 'saveBeforeResponse', {
-        value: false,
-        writable: true
-    });
-
-    Object.defineProperty(handler, 'i18n', {
-        value: i18n,
-        writable: true
-    });
-
-    Object.defineProperty(handler, 'locale', {
-        value: undefined,
-        writable: true
-    });
-
-    Object.defineProperty(handler, 'resources', {
-        value: undefined,
-        writable: true
-    });
-
-    Object.defineProperty(handler, 'registerHandlers', {
-        value: function() {
-            RegisterHandlers.apply(handler, arguments);
-        },
-        writable: false
-    });
-
-    Object.defineProperty(handler, 'execute', {
-        value: function() {
-            HandleLambdaEvent.call(handler);
-        },
-        writable: false
-    });
-
-    handler.registerHandlers(responseHandlers);
-
-    return handler;
+  const newEvent = event;
+  if (!newEvent.session) {
+    newEvent.session = {attributes: {}};
+  } else if (!event.session.attributes) {
+    newEvent.session.attributes = {};
+  }
+
+  const handler = new AlexaRequestEmitter();
+  handler.setMaxListeners(Infinity);
+
+  Object.defineProperty(handler, '_event', {
+    value: newEvent,
+    writable: false,
+  });
+
+  Object.defineProperty(handler, '_context', {
+    value: context,
+    writable: false,
+  });
+
+  Object.defineProperty(handler, '_callback', {
+    value: callback,
+    writable: false,
+  });
+
+  Object.defineProperty(handler, 'state', {
+    value: null,
+    writable: true,
+    configurable: true,
+  });
+
+  Object.defineProperty(handler, 'appId', {
+    value: null,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'response', {
+    value: {},
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'dynamoDBTableName', {
+    value: null,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'saveBeforeResponse', {
+    value: false,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'i18n', {
+    value: i18n,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'locale', {
+    value: undefined,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'resources', {
+    value: undefined,
+    writable: true,
+  });
+
+  Object.defineProperty(handler, 'registerHandlers', {
+    value: function registerHandlers(...args) {
```

Comment:
Is there a reason why you don't use an arrow function? (this question also applies to the next 2 property definitions)